src/main/java/org/marketdesignresearch/mechlib/instrumentation/MipInstrumentation.java

```diff
@@ -1,12 +1,13 @@
 package org.marketdesignresearch.mechlib.instrumentation;
 
+import java.util.ArrayList;
+import java.util.List;
+
+import org.marketdesignresearch.mechlib.core.Allocation;
+
 import edu.harvard.econcs.jopt.solver.IMIP;
 import edu.harvard.econcs.jopt.solver.IMIPResult;
 import lombok.EqualsAndHashCode;
-import org.marketdesignresearch.mechlib.core.Allocation;
-
-import java.util.ArrayList;
-import java.util.List;
 
 @EqualsAndHashCode
 public class MipInstrumentation {
```

Comment:
@islerfab Currently you have a fixed enum of MIP purposes. Now I might want to implement other purposes (i.e. ML Allocation Inferrer and Support Vector). Can we implement the MIP purpose in a more flexible way such that a new AuctionPhase or Bidder Strategy that uses MIPs must not change this class here?

Reply:
Of course, we can and we should. Do you have a proposition on how it serves your purpose best?

Reply:
Unfortunately I don't have a clean way to do this. Strings may be not unique so another solution would be to use sub-interfaces of a common supertype. However, I think both ways are not very clean.

Reply:
Do you have a solution for this issue? I think using Strings and Constants would maybe the best and flexible solution.

Reply:
@islerfab This is the only open question left in this thread. Otherwise as suggested by you we might merge this thread and start a new one for further discussions.

Reply:
I thought I have responded to that, I'm sorry... I also think that using strings is reasonable here... At least for now. If this starts to be complex with many categories of purposes, we want to rethink it, but I suggest to proceed the way you're suggesting.

Reply:
I changed this in an mlca branch. Now the purpose is a String, but for now I just use the .name() of the enum as identifier.
Another question came up when I did this. Should the purpose not be more specific? Like:
"payment.euclideanNorm"
"payment.BlockingAllocation"
"allocation.XOR"

Otherwise you can not really use it to work just on one specific MIP when you use the same MIPInstrumentation for all MIPs.

Reply:
String is fine with me, and I like the idea of potentially more specific purposes.